### DISCUSSION TITLE
DataGrid ColumnChooser option + Load/Save State (or Layout)

### DISCUSSION THREAD

**Discussion by JurjendeGroot** [microsoft/fluentui-blazor#2853] [Ideas]
We would like to be able to use a column chooser within the DataGrid, so you can at runtime add / remove columns to the grid?
    Like a visibility toggle for the various available columns in the grid where you could also change the location of the column, so move the OrderNumber column as the first to the last column.
    
As the user has then customized his view of the data and maybe sorted some column(s) it would be nice if that 'layout (or state)' could be saved to a database or local storage so it could be used to restore the grid to this state the next time the user visits.

Are there any plans for these kind of features on the roadmap of FluentUI ?

![image](https://github.com/user-attachments/assets/0526c215-0d6e-4194-b954-80b9b24eec22)

One can drag columns up or down by clicking-and-holding on the little dotted bar on the right of the columnames.
One can click the checkbox on the left to set visibility of the column on or off.
Then next one can make buttons/menu with which the user can save or load the state (which columns shown, the column sequence, current sorting, grouping, etc.) to a Json and store that in a db or anywhere else.

**Comment by vnbaaij** [comment]
Hi,

We currently have no plans to add this kind of functionality. The current grid is based on QuickGrid and uses the Fluent Ui `<fluent-data-grid-...>` components to render  the outpit. For a future version (4.11) we are changing this to use HTML tables again (like original QuickGrid). 
We are doing this because it will take a while for new web components to become available for us. We are building the next version of the library on these new web components and do not want to wait until everything is there and we also do not want to release without a datagrid. We want to wait till at least more is clear on what the new web components are going to offer and we are trying not to deviate too much from what they delivered earlier. Hope this makes sense.

**Reply by JurjendeGroot** [reply]
Hi Vincent,

Thanks for your response.
